mbrl/util/replay_buffer.py

```diff
@@ -520,25 +524,28 @@ def add(
         action: np.ndarray,
         next_obs: np.ndarray,
         reward: float,
-        done: bool,
+        terminated: bool,
+        truncated: bool,
     ):
-        """Adds a transition (s, a, s', r, done) to the replay buffer.
+        """Adds a transition (s, a, s', r, terminated) to the replay buffer.
 
         Args:
             obs (np.ndarray): the observation at time t.
             action (np.ndarray): the action at time t.
             next_obs (np.ndarray): the observation at time t + 1.
             reward (float): the reward at time t + 1.
-            done (bool): a boolean indicating whether the episode ended or not.
+            terminated (bool): a boolean indicating whether the episode ended in a terminal state.
+            truncated (bool): a boolean indicating whether the episode ended prematurely.
         """
         self.obs[self.cur_idx] = obs
         self.next_obs[self.cur_idx] = next_obs
         self.action[self.cur_idx] = action
         self.reward[self.cur_idx] = reward
-        self.done[self.cur_idx] = done
+        self.terminated[self.cur_idx] = terminated
+        self.truncated[self.cur_idx] = truncated
 
         if self.trajectory_indices is not None:
-            self._trajectory_bookkeeping(done)
+            self._trajectory_bookkeeping(terminated)
```

Comment:
Are we sure about this? For this book keeping, it seems to me that we need the old meaning of `done`, that is, `terminated or truncated`. Otherwise, the info for a truncated trajectory will not be saved in the buffer, or be closed either, and the buffer will try to append transitions from the next trajectory to the truncated one. This doesn't affect sampling transitions, but it does affect the code used to sample full trajectories.

Reply:
Correct, fixing this

Reply:
The issue is that the trajectory is ["closed"](https://github.com/facebookresearch/mbrl-lib/blob/3c6553a43e8fea37b0695dacab91e3fefb147ad7/mbrl/util/replay_buffer.py#L502) only when `terminated = True`. The closing method is the only part of the code that saves the start/end indices to the list of trajectories that can be sampled, and the only method that [resets](https://github.com/facebookresearch/mbrl-lib/blob/3c6553a43e8fea37b0695dacab91e3fefb147ad7/mbrl/util/replay_buffer.py#L515) the start of the last trajectory, which makes it so that the next new trajectory is tracked correctly.

Reply:
Ah, that makes sense. Fixed now.